docs/spec/v1alpha1/gitrepositories.md

```diff
@@ -0,0 +1,216 @@
+# Git Repositories
+
+The `GitReposiory` API defines a source for artifacts coming from Git. 
+
+## Specification
+
+Git repository spec:
+
+```go
+// GitRepositorySpec defines the desired state of GitRepository
+type GitRepositorySpec struct {
+	// +kubebuilder:validation:Pattern="^(http|https|ssh)://"
+
+	// The repository URL, can be a HTTP or SSH address.
+	Url string `json:"url"`
+
+	// The secret name containing the Git credentials
+	// +optional
+	SecretRef *v1.LocalObjectReference `json:"secretRef,omitempty"`
+	
+	// The git reference to checkout and monitor for changes, defaults to master branch.
+	// +optional
+	Reference *GitRepositoryRef `json:"ref,omitempty"`
+
+	// The interval at which to check for repository updates.
+	Interval metav1.Duration `json:"interval"`
+}
+
+// GitRepositoryRef defines the git ref used for pull and checkout operations
+type GitRepositoryRef struct {
+	// The git branch to checkout, defaults to master.
+	// +optional
+	Branch string `json:"branch"`
+
+	// The git tag to checkout, takes precedence over branch.
+	// +optional
+	Tag string `json:"tag"`
+
+	// The git tag semver expression, takes precedence over tag.
+	// +optional
+	SemVer string `json:"semver"`
+
+	// The git commit sha to checkout, if specified branch and tag filters will be ignored.
+	// +optional
+	Commit string `json:"commit"`
+}
+```
+
+Git repository status:
+
+```go
+// GitRepositoryStatus defines the observed state of GitRepository
+type GitRepositoryStatus struct {
+	// +optional
+	Conditions []RepositoryCondition `json:"conditions,omitempty"`
+
+	// LastUpdateTime is the timestamp corresponding to the last status
+	// change of this repository.
+	// +optional
+	LastUpdateTime *metav1.Time `json:"lastUpdateTime,omitempty"`
+
+	// URI for the artifacts of the last successful repository sync.
+	// +optional
+	Artifacts string `json:"artifacts,omitempty"`
+}
+```
+
+Git repository status conditions:
+
+```go
+// RepositoryCondition contains condition information for a repository
+type RepositoryCondition struct {
+	// Type of the condition, currently ('Ready').
+	Type RepositoryConditionType `json:"type"`
+
+	// Status of the condition, one of ('True', 'False', 'Unknown').
+	Status corev1.ConditionStatus `json:"status"`
+
+	// LastTransitionTime is the timestamp corresponding to the last status
+	// change of this condition.
+	// +optional
+	LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty"`
+
+	// Reason is a brief machine readable explanation for the condition's last
+	// transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
```

Comment:
Given this field is a machine readable explanation I would expect the possible values to be registered as constants. An advantage of this from a user perspective is that they will be predictable and easy to document, to us it gives a slight advantage of easy typing.

Reply:
Addressed in https://github.com/fluxcd/source-controller/commit/8cd8d8f6f23798e81baf49817741647aa4315358